R/ForecastRegrTask.R

```diff
@@ -0,0 +1,58 @@
+#' @title Create a task for univariate forecasting
+#'
+#' @rdname Task
+#'
+#' @description Creates a task for univariate forecasting learners
+#'
+#' @export
+makeForecastRegrTask = function(id = deparse(substitute(data)), data, target,
+  weights = NULL, blocking = NULL, frequency = 1L, fixup.data = "warn",
+  check.data = TRUE) {
+  assertString(id)
+  assertClass(data, "xts")
```

Comment:
If there is always going to be exactly one date column, it would make sense to have a `date` argument, check for uniqueness and type, and then use
```R
data = data[order(data[[date]]), , drop = FALSE]
```

Maybe one could also convert the data from the input `data.frame` to an `xts` object and back if the sanity checks are more involved. (This might be slow if a lot of data is involved, but probably still negligible relative to ML operations?)

Probably @berndbischl has a stronger opinion on this than I do